python/GafferUI/GraphBookmarksUI.py

```diff
@@ -127,6 +144,55 @@ def popupFindBookmarkMenu( editor ) :
 
 	__findBookmark( editor )
 
+def setNumericBookmark( editor, numericBookmark ) :
+
+	if not isinstance( editor, GafferUI.GraphEditor ) :
+		return False
+
+	selection = editor.scriptNode().selection()
+	node = None
+	if len( selection ) == 1 :
+		node = selection[0]
+	else :
+		backdrops = [n for n in selection if isinstance( n, Gaffer.Backdrop )]
```

Comment:
Handling backdrops is a little weird because of them selecting all contained nodes automatically. I think this works even though it looks a little odd.

Reply:
I hadn't even thought about that - nice one. This makes this all more useful as an alternative to that internal ticket about "backdrop only bookmarks". If we wanted to be totally consistent I suppose we should check that all the other nodes we ignored are actually on the backdrop, but I think this is plenty good enough...